app/src/main/java/com/infomaniak/mail/data/models/Folder.kt

```diff
@@ -71,10 +70,17 @@ class Folder : RealmObject {
     var threads: RealmList<Thread> = realmListOf()
     @Transient
     var messages: RealmList<Message> = realmListOf()
+
+    /**
+     * List of old Messages UIDs of this Folder that we need to fetch.
+     * When first opening the Folder, we get the full list of UIDs, and we store it.
+     * Then, we'll be able to go through it as we want to fetch the old Messages.
+     */
     @Transient
-    var remainingOldMessagesToFetch: Int = DEFAULT_REMAINING_OLD_MESSAGES_TO_FETCH
+    var oldMessagesUidsToFetch = realmListOf<Int>()
```

Comment:
Check if we receive 50 new Threads when fetching Messages.
If not, continue looping deeper into `oldMessagesUidsToFetch` until 50 new Threads have been added in the Folder.

Reply:
Done.

The limit has been fixed at `PAGE_SIZE / 2` _(25)_.